examples/jsm/renderers/webgpu/WebGPURenderPipeline.js

```diff
@@ -700,10 +639,10 @@ class WebGPURenderPipeline {
 			const type = nodeAttribute.type;
 
 			const geometryAttribute = geometry.getAttribute( name );
-			const bytesPerElement = geometryAttribute.array.BYTES_PER_ELEMENT;
 
-			const format = this._getVertexFormat( type, bytesPerElement );
+			const format = this._getVertexFormat( geometryAttribute );
 
+			const bytesPerElement = geometryAttribute.array.BYTES_PER_ELEMENT;
 			let arrayStride = geometryAttribute.itemSize * bytesPerElement;
```

Comment:
Given an InterleavedBufferAttribute we'll need to use its associated [`.stride`](https://threejs.org/docs/?q=Interleaved#api/en/core/InterleavedBuffer.stride) for WebGPU's `arrayStride` property, not just infer stride from the element size. If we have a normal `BufferAttribute`, then I think layouts like `uint16x3` are just not allowed. GLTFLoader should create InterleavedBufferAttribute automatically here.

Reply:
Deeper improvements to the interleaved vertex layout have been suggested — https://github.com/mrdoob/three.js/issues/17089. Those would be helpful for optimization purposes, but aren't necessary to get things working in WebGPU.

Reply:
The `.stride` is for the interleaved object, not for this specific field so I'm not sure how that's useful here. For example if the vertex is `16x3 position + 16x3 normal + 32x4 boneWeights + 1x4 boneIndices` then when we're here for `position` then `.stride = 16`

Reply:
> ... when we're here for position then .stride = 16

Not necessarily — stride is entirely dependent on the vertex layout chosen by the author. We can't infer the position attribute's stride without reading the interleaved buffer object. Position may be interleaved with normals, or interleaved with vertex data from another mesh, or only "interleaved" with its own padding. Each case would give a different stride for an otherwise identical InterleavedBufferAttribute.

Reply:
We know it's 16x3 as the type is "`vec3`", and the arraybuffer is `Int16Array` or `Uint16Array`. We also know that WebGPU doesn't support 16x3 so it really needs to be 16x4. If that fails then it wasn't compatible with WebGPU in the first place. 

The actual stride in bytes from the gLTF file is already correct.

Reply:
I don't know if I'm making sense.

My point is

1. We know the code is asking for "vec3" of 16bit values
2. We know WebGPU doesn't support 3x16
3. We don't know at this level if changing it 4x16 is valid. There is no data at this level that will let us derive if it's valid to change 3x16 to 4x16 or not.
4. We do know that not changing it from 3x16 to 4x16 is incompatible with WebGPU. So, we just change it and cross our fingers.

I'm not sure what info you could add that would solve point 3 above. For example "vec3" of 16bit values with a stride of 8bytes does not tell you that it's valid to change this from 3x16 to 4x16 because the interleaved format might be 

```
struct {
  normal: vec3<int16>
  something: int8
  somethingElse: uint8
}
```

Reply:
Agreed on all of that, and selecting 4x16. But to fully support KHR_mesh_quantization or arbitrary interleaved data we also need to give WebGPU the right "arrayStride", which the line of code above does not do. It could be 8 bytes, 16 bytes, or even 1024 bytes ... we can only get that information from buffer.stride.

Reply:
The arrayStride, when from KHR_mesh_quantization is already correct given the spec requires it to be a multiple of 4.  

If it's from a user created InterleavedBufferAttribute(sp?) (not from gLTF) we could print an error `arrayStride must be multiple of 4`

Reply:
This line of code, above, is incorrect:

```js
let arrayStride = geometryAttribute.itemSize * bytesPerElement;
```

In addition to being a multiple of four, it must also match the actual stride in the actual data, which this line does not check. It should be reading `buffer.stride` for that. We cannot just pick the smallest convenient multiple of four.

Reply:
Right: So the reason it's working for both of your files is that even though you passed `--vertex-layout separate` for one of the files, at least when it gets the that part of the code, it takes the `if` path

```
  arrayStride = geometryAttribute.data.stride * bytesPerElement;
```

Reply:
In anycase, I wasn't planning on pursuing this PR. I was just sharing it because it worked for your two files and thought it might be useful info for whoever decides how to deal with this issue in the three.js WebGPURenderer

Reply:
Ok, yes — let's call this resolved, thanks! I'll just leave a note in the main thread that `buffer.stride` should probably be considered in the final PR.